javagen/src/main/java/com/azure/autorest/template/ClientMethodTemplate.java

```diff
@@ -221,98 +303,121 @@ protected static void applyParameterTransformations(JavaBlock function, ClientMe
+                    addedConversion = true;
+                } else if (parameterClientType instanceof ListType) {
+                    boolean alreadyNullChecked = clientMethod.getRequiredNullableParameterExpressions()
+                        .contains(parameterName);
+                    String expression;
+                    if (alwaysNull) {
+                        expression = "null";
+                    } else if (!parameter.getExplode()) {
+                        CollectionFormat collectionFormat = parameter.getCollectionFormat();
+                        if (((ListType) parameterClientType).getElementType() instanceof EnumType) {
+                            // EnumTypes should provide a toString implementation that represents the wire value.
+                            // Circumvent the use of JacksonAdapter and handle this manually.
+
+                            // If the parameter is null, the converted value is null.
+                            // Otherwise, convert the parameter to a string, mapping each element to the toString
+                            // value, finally joining with the collection format.
+                            if (alreadyNullChecked) {
+                                expression =
+                                    parameterName + ".stream()\n" +
+                                    "    .map(value -> Objects.toString(value, \"\"))\n" +
+                                    "    .collect(Collectors.joining(\"" + collectionFormat.getDelimiter() + "\"))";
                             } else {
-                                expression = String.format("Base64Util.encodeToString(%s)", parameterName);
+                                expression =
+                                    "(" + parameterName + " == null) ? null : " + parameterName + ".stream()\n" +
+                                        "    .map(value -> Objects.toString(value, \"\"))\n" +
+                                        "    .collect(Collectors.joining(\"" + collectionFormat.getDelimiter() + "\"))";
                             }
-                            function.line("%s %s = %s;", parameterWireType, parameterWireName, expression);
                         } else {
-                            String expression;
-                            if (alwaysNull) {
-                                expression = "null";
-                            } else {
-                                expression = String.format("Base64Url.encode(%s)", parameterName);
-                            }
-                            function.line("%s %s = %s;", parameterWireTypeName, parameterWireName, expression);
+                            expression = String.format(
+                                "JacksonAdapter.createDefaultSerializerAdapter().serialize%s(%s, CollectionFormat.%s)",
+                                settings.shouldUseIterable() ? "Iterable" : "List", parameterName,
```

Comment:
Please do be careful that they are indeed equivalent. Else it could be nasty bug.